File: targets/dotnet/fastendpoints/suite/BaselineTests.cs
using FastEndpoints.Testing;
using RequestBench.Suite;
using Xunit;

namespace RequestBench.FastEndpointsTarget.Suite;

/// <summary>
/// baseline: the dispatch floor, with no serialization in the way.
///
/// The one endpoint in the corpus that answers a literal. Its whole contract is the string
/// and the content type, and the content type is the half a test gets wrong: a target that
/// answers "Hello, World!" as application/json has passed the body and failed the endpoint.
/// The floor checks the kind of body before the body for that reason.
/// </summary>
public sealed class BaselineTests(TargetApp app) : TestBase<TargetApp>
{
    // rb:test baseline.plaintext
    [Fact]
    public async Task The_plaintext_route_answers_a_literal_as_text()
    {
        Ask ask = Plan.For("baseline.plaintext");

        using HttpResponseMessage response = await app.Send(ask);

        await Floor.AssertAsync(response, ask);
        Assert.Equal("text/plain", response.Content.Headers.ContentType?.MediaType);
    }
}

File: targets/java/helidon-se/src/test/java/rb/helidon/BaselineTests.java
package rb.helidon;

import static org.junit.jupiter.api.Assertions.assertTrue;

import io.helidon.webclient.http1.Http1Client;
import org.junit.jupiter.api.Test;

/**
 * baseline: the dispatch floor, with no serialization in the way.
 *
 * <p>The one endpoint in the corpus that answers a literal. Its whole contract is the string
 * and the content type, and the content type is the half a test gets wrong: a target that
 * answers "Hello, World!" as application/json has passed the body and failed the endpoint.
 * The floor checks the kind of body before the body for that reason.
 */
class BaselineTests extends HelidonSuite {

  BaselineTests(Http1Client client) {
    super(client);
  }

  // rb:test baseline.plaintext
  @Test
  void the_plaintext_route_answers_a_literal_as_text() {
    Planned.Ask a = Planned.ask("baseline.plaintext");

    Floor.Answer answer = send(a);

    Floor.check(a, answer);
    assertTrue(answer.contentType().startsWith("text/plain"));
  }
}
